cli/execute/execute.go

```diff
@@ -156,11 +156,16 @@ func execute(cmdArgs []string) error {
 		return err
 	}
 	log.Debugf("Uploaded inputs in %s", time.Since(stageStart))
-	actionStr, err := digest.ResourceNameFromProto(arn).DownloadString()
-	if err != nil {
-		log.Debugf("Failed to compute action resource name: %s", err)
+	acrn, err := digest.ResourceNameFromProto(arn).CheckCAS()
```

Comment:
Can you add helpers to create the appropriate resource type to avoid the caller having to make an extra call to either CheckCAS or checkAC? i.e. `digest.CASResourceNameFromProto` and `digest.ACResourceNameFromProto`.

Reply:
Added